Split/Network/Streaming/SseHandler.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  SseHandler.swift
+//  Split
+//
+//  Created by Javier L. Avrudsky on 01/09/2020.
+//  Copyright © 2020 Split. All rights reserved.
+//
+
+import Foundation
+
+protocol SseHandler {
+    func handleIncomingMessage(message: [String: String])
+}
+
+class DefaultSseHandler: SseHandler {
+    private let kDataField = "data"
+    let notificationProcessor: SseNotificationProcessor
+    let notificationParser: SseNotificationParser
+    let notificationManagerKeeper: NotificationManagerKeeper
+    let broadcasterChannel: PushManagerEventBroadcaster
+
+    init(notificationProcessor: SseNotificationProcessor,
+         notificationParser: SseNotificationParser,
+         notificationManagerKeeper: NotificationManagerKeeper,
+         broadcasterChannel: PushManagerEventBroadcaster) {
+
+        self.notificationProcessor = notificationProcessor
+        self.notificationParser = notificationParser
+        self.notificationManagerKeeper = notificationManagerKeeper
+        self.broadcasterChannel = broadcasterChannel
+    }
+
+    func handleIncomingMessage(message: [String: String]) {
+        guard let data = message[kDataField] else {
+            return
+        }
+
+        guard let incomingNotification = notificationParser.parseIncoming(jsonString: data) else {
+            return
+        }
+
+        switch incomingNotification.type {
+        case .control:
+            print("TODO: handle control here")
```

Comment:
just as a suggestion, consider handling Control messages similar than Occupancy messages, by passing them to the NotificationManagerKeeper (e.g., `notificationManagerKeeper.handleIncomingControlEvent(notification: notification)`), since the NotificationManagerKeeper is already emitting PUSH_SUBSYSTEM_UP/DOWN events due to occupancy messages.